File: BinToDec.cs

/*
 * Demonstrates an implementation of converting a binary number to a decimal
 * number in Java.
 *
 * Auther: Josh Brown
 * Date: 2/1/2021
 *
 */

using System;
public class MainClass {
    static void Main(string[] args) {
        string binary = "100001000101111111101101"; // 8675309 as binary
        Console.WriteLine(BinToDec(binary));
    }

    /*
     * Returns the decimal equivalent of a binary string
     * Input: string representing a binary number
     * Output: int decimal equivalent of input
     */
    static int BinToDec(string binary) {
        int dec = 0;
        for(int i = binary.Length; i > 0; i--) {
            int val = (int)binary[i-1] - '0';
            dec += val*pow(2,binary.Length-i);
        }
        return dec;
    }

    /*
     * Returns a raised to the power b
     * Input: integer a, integer b
     * Output: integer value representing a^b
     */
    static int pow(int a, int b) {
        int result = 1;
        for(; b > 0; b--) {
            result *= a;
        }
        return result;
    }
}

File: BinToDec.py

'''

Demonstrates an implementation of converting a binary number to a decimal number
in Python.

Author: Josh Brown
Date: 2/1/2021

'''

# Returns the decimal equivalent of binary string
# Input: string representing a binary number
# Output: decimal equivalent of input
def BinToDec(bin):
    dec = 0
    for i in range(len(bin), 0, -1):
        val = int(bin[i-1])
        dec = dec + val*2**(len(bin)-i)
    return dec

# Main - passes binary equivalent of "8675309" to BinToDec function and prints
# result
def main():
    bin = "100001000101111111101101"
    print(BinToDec(bin))

if __name__ == '__main__':
    main()
